yarn-project/end-to-end/src/e2e_multi_transfer.test.ts

```diff
@@ -0,0 +1,119 @@
+import { AztecNodeService } from '@aztec/aztec-node';
+import { AztecRPCServer } from '@aztec/aztec-rpc';
+import { AztecAddress, Contract, Fr, Wallet } from '@aztec/aztec.js';
+import { DebugLogger } from '@aztec/foundation/log';
+import { PrivateTokenAirdropContract } from '@aztec/noir-contracts/types';
+import { MultiTransferContract } from '@aztec/noir-contracts/types';
+import { AztecRPC } from '@aztec/types';
+
+import { expectsNumOfEncryptedLogsInTheLastBlockToBe, setup } from './fixtures/utils.js';
+
+describe('multi-transfer payments', () => {
+  const numberOfAccounts = 6;
+
+  let aztecNode: AztecNodeService | undefined;
+  let aztecRpcServer: AztecRPC;
+  let wallet: Wallet;
+  let accounts: AztecAddress[];
+  let logger: DebugLogger;
+  let ownerAddress: AztecAddress;
+  const recipients: AztecAddress[] = [];
+
+  let zkTokenContract: PrivateTokenAirdropContract;
+  let multiTransferContract: MultiTransferContract;
+
+  beforeEach(async () => {
+    ({ aztecNode, aztecRpcServer, accounts, logger, wallet } = await setup(numberOfAccounts + 1)); // 1st being the `owner`
+    ownerAddress = accounts[0];
+
+    for (let i = 1; i < accounts.length; i++) {
+      const account = accounts[i];
+      recipients.push(account);
+    }
+  }, 100_000);
+
+  afterEach(async () => {
+    await aztecNode?.stop();
+    if (aztecRpcServer instanceof AztecRPCServer) {
+      await aztecRpcServer?.stop();
+    }
+  }, 30_000);
+
+  const deployZkTokenContract = async (initialBalance: bigint, owner: AztecAddress) => {
+    logger(`Deploying zk token contract...`);
+    zkTokenContract = await PrivateTokenAirdropContract.deploy(wallet, initialBalance, owner).send().deployed();
+    logger(`zk token contract deployed at ${zkTokenContract.address}`);
+  };
+
+  const deployMultiTransferContract = async () => {
+    logger(`Deploying multi-transfer contract...`);
+    multiTransferContract = await MultiTransferContract.deploy(wallet).send().deployed();
+    logger(`multi-transfer contract deployed at ${multiTransferContract.address}`);
+  };
+
+  const expectBalance = async (tokenContract: Contract, owner: AztecAddress, expectedBalance: bigint) => {
+    const [balance] = await tokenContract.methods.getBalance(owner).view({ from: owner });
+    logger(`Account ${owner} balance: ${balance}`);
+    expect(balance).toBe(expectedBalance);
+  };
+
+  it('12 transfers per transactions should work', async () => {
+    const initialNote = 1000n;
+
+    logger(`Deploying zk token contract...`);
+    await deployZkTokenContract(initialNote, ownerAddress);
+
+    logger(`Deploying multi-transfer contract...`);
+    await deployMultiTransferContract();
+
+    // owner: 1000 => [100, 200, 300, 400]
+    logger(`self batchTransfer()`);
+    const batchTransferTx = zkTokenContract.methods
```

Comment:
@iAmMichaelConnor - should we have a library method for splitting your 1 utxo note into n notes of certain values? Seems useful to work with the UTXO state!

Reply:
I'm not sure. You'd have to specify the n values and the n owners of the note. And it only really applies to the `ValueNote`. I'd be in favour of not abstracting it yet, and seeing how the community cope with the utxo syntax (rather than trying to hide it completely).